### PR TITLE
[JBPM-10242] Allow the possibility of disabling linear search for removeJob and getTimerByName operations

### PR DESCRIPTION
**Pull request by fjtirado**
Fix https://issues.redhat.com/browse/JBPM-10242

### CODE REVIEW

**Comment by gmunozfe** [comment]
jenkins do fdb

**Review by gmunozfe — APPROVED** [review]
Looks good to me! Thanks for your work @krisv , it's a weird scenario and this should make the code more robust by skipping the searching also in case there's no exception caught.

File: jbpm-services/jbpm-services-ejb/jbpm-services-ejb-timer/src/main/java/org/jbpm/services/ejb/timer/EJBTimerScheduler.java [hunk 1]
```diff
@@ -271,15 +278,22 @@ public boolean removeJob(JobHandle jobHandle, Timer ejbTimer) {
...
                 }
                 return true;
+            } else {
+                logger.warn("No TimerHandle found for {}: {}", ejbHandle, ejbTimerHandle);
             }
         } else {
             logger.warn("No timerJobInstance available for {}", ejbHandle);
         }
 
+        if (!searchIfFailed) {
```
Comment (fjtirado) [inline jbpm-services/jbpm-services-ejb/jbpm-services-ejb-timer/src/main/java/org/jbpm/services/ejb/timer/EJBTimerScheduler.java:292]: Is this change really needed? ( I have not gone through latest traces yet), but I feel this will prevent searching in case the search migh be needed ( to be honest, I do not completely understand the need for the search, so I was being conservative in the first commit and only disable it when it is clear that the times is there and the cancel has failed)

Comment (fjtirado) [reply jbpm-services/jbpm-services-ejb/jbpm-services-ejb-timer/src/main/java/org/jbpm/services/ejb/timer/EJBTimerScheduler.java:292]: After reviewing the logs, this change makes sense. Probably we need more...

**Comment by fjtirado** [comment]
jenkins run fdb

**Comment by fjtirado** [comment]
jenkins run fdb

**Comment by fjtirado** [comment]
jenkins run fdb

**Comment by fjtirado** [comment]
jenkins run fdb

**Comment by gmunozfe** [comment]
jenkins do fdb

**Review by gmunozfe — APPROVED** [review]
Looks good to me.
Works like a charm: tested regression scenarios with jbpm-playground tests for timers (complex scenarios) with and without cache and enabling/disabling flags.
Amazing work @fjtirado !

**Comment by fjtirado** [comment]
thanks @gmunozfe
